### PR TITLE
Upgrade Mapbox SDK 2.0

### PR DESCRIPTION
**Pull request by JorgeQuevedoC**
This PR upgrades the Mapbox Navigation SDK for iOS to 2.1.0 and 2.0.0 for Android.

The Android Mapbox Navigation SDK for version 2.X.X no longer supports the drop in UI as mentioned here: https://docs.mapbox.com/android/navigation/guides/migrate-to-v2/#navigationview-was-removed. So this PR implements a fully customizable UI for Android.

Everything will function the same except the prop endOfRouteFeedback will not work Android for now. If we want to support the feedback functionality we will need to create our own UI for this as this was removed from the drop in UI Mapbox supported for Android in previous versions.

### CODE REVIEW

**Comment by JorgeQuevedoC** [comment]
@joe307bad Any idea on how to run these changes and see the outcome ?

**Comment by joe307bad** [comment]
Thanks @JorgeQuevedoC, this looks pretty straight forward. I am going to find some time in the next few days to devise a testing strategy. Feel free to share any ideas here.

**Comment by JorgeQuevedoC** [comment]
I think that the best idea, as you also mentioned, is to create some examples within the same repo

**Comment by joe307bad** [comment]
Yep! Here is a [good example](https://github.com/ezranbayantemur/react-native-rtmp-publisher/tree/master/example) an example app inside a react native package. We should work on adding a folder like this in this package.

**Comment by JorgeQuevedoC** [comment]
Hey I have just updated the PR with the Examples folder already set up! 

image

**Comment by joe307bad** [comment]
This is really good work @JorgeQuevedoC, thanks for putting this together. I am going to find some time today to run this locally burt the screenshot definitely goes a long way. A small request, can we move the example app back one directory so its in `/example` not `/example/BasicApp`?

**Comment by JorgeQuevedoC** [comment]
@joe307bad Sure thing! I will push as soon as it is done and tested

**Comment by JorgeQuevedoC** [comment]
Hi @joe307bad, Changes are already push into the PR with just 1 example folder

**Comment by gciluffo** [comment]
Yo @JorgeQuevedoC! Im going to be helping out a bit with this library too when I have free time. Thanks for taking the initiative to make a PR. A couple things on this PR:

- We need to make sure the iOS build version is set to 11. Thats the minimum build version required for this new version of MapBox as seen here: https://docs.mapbox.com/ios/navigation/guides/#requirements. I would also recommend taking a look at the migration guide to version 2 in case there is anything we missed: https://docs.mapbox.com/ios/navigation/guides/migrate-to-v2/#upgrading-mapboxnavigation-from-v1x-to-v200
- In the diff it seems like theres a lot of permission changes on a lot of the files i.e. `100644 → 100755`. Im not sure if that intentional or not but if its not can we revert those files back to the way they were so the diff is easier to read for future pruposes?
- With adding the examples directory we should also add that to the exclude property in tsconfig
- Lastly if you are comfortable with it, it would also be nice to upgrade the sdk that Android is using to v2.

**Comment by JorgeQuevedoC** [comment]
Hi @gciluffo, great feedback and glad to see more people involved on this. Tomorrow I will start reviewing the first 3 bullets; but, I am not very familiar to the process to update Android SDK. I would like to be involved but some help would be much appreciated

**Comment by gciluffo** [comment]
@JorgeQuevedoC No worries, I can take a look at upgrading the Android SDK.

File: example/ios/BasicApp/Info.plist [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	<key>CFBundleShortVersionString</key>
+	<string>1.0</string>
+	<key>CFBundleSignature</key>
+	<string>????</string>
+	<key>CFBundleVersion</key>
+	<string>1</string>
+	<key>LSRequiresIPhoneOS</key>
+	<true/>
+	<key>MBXAccessToken</key>
+	<string>sk.eyJ1Ijoiam9yZ2VxdWV2ZWRveCIsImEiOiJja29ib3oycWIyN2Z3MnZvbmc0eGttOTI0In0.Wsd8kBSIh6RkhivVUWm3cw</string>
```
Comment (gciluffo) [inline example/ios/BasicApp/Info.plist:28]: This should be the public key not the secret key.

Comment (gciluffo) [reply example/ios/BasicApp/Info.plist:28]: Actually this is probably fine just an example project.

Comment (JorgeQuevedoC) [reply example/ios/BasicApp/Info.plist:28]: Good catch, thanks

**Comment by JorgeQuevedoC** [comment]
@gciluffo I think we are ready on iOS side. Please correct me if wrong and let me know if something else is required

**Comment by gciluffo** [comment]
@JorgeQuevedoC I think we just need to update [this](https://github.com/homeeondemand/react-native-mapbox-navigation/pull/66/files#diff-b565d99971e85948f9a6bdb1e2763530c581e225883e0073a7c7ef383eec776aR42) to version 11. But other than that everything else looks good.

I am currently working on upgrading the SDK for Android on your fork. It looks like theres some compatibility issues with using Gradle 7 which is what the example project is using. So hopefully will have that working along with the Mapbox SDK upgrade before the week is over.

**Comment by JorgeQuevedoC** [comment]
@gciluffo I don't get where should I update the version

**Comment by gciluffo** [comment]
@JorgeQuevedoC I should have a PR for the android upgrade within a couple days. One thing I noticed when testing the iOS changes is that nothing happens when tapping the X to cancel navigation. The event from the swift code propagates up to React Native properly but the navigation doesnt stop when tapping it.

**Comment by JorgeQuevedoC** [comment]
@gciluffo those are great news. I didn't change that functionality but I can review it. So it must execute the function passed through props and after that stop the navigation no matter what, right?

**Comment by gciluffo** [comment]
@JorgeQuevedoC Correct. It looks like its an outstanding issue this https://github.com/homeeondemand/react-native-mapbox-navigation/issues/34. However it seems like it existed well before we upgraded iOS SDK. No worries if you dont want to look address in this PR.

**Comment by JorgeQuevedoC** [comment]
@gciluffo I can address it. Probably I can push that change tomorrow!

**Comment by gciluffo** [comment]
@JorgeQuevedoC Lets just address it in another PR. I didnt implement cancelling a route in the android PR I submitted to your fork. That way the functionality between android and ios are the same for now. Also its not clear what the best approach to cancelling the route is either because in this implementation a user cannot create another route once they cancel it.

**Comment by JorgeQuevedoC** [comment]
@gciluffo understood. Let's address it later then

**Comment by JorgeQuevedoC** [comment]
@gciluffo let's give it a final review

**Comment by gciluffo** [comment]
@JorgeQuevedoC Are able to check the `Allow edits by maintainers` checkbox for this PR?

**Comment by JorgeQuevedoC** [comment]
@gciluffo it was checked

image

**Comment by gciluffo** [comment]
@JorgeQuevedoC I am still having permission issues pushing to your fork directly. I just went ahead and opening another pull request for your fork if you can a look at it. And then I think we should be good to go.

**Comment by JorgeQuevedoC** [comment]
@gciluffo done!

**Comment by gciluffo** [comment]
@JorgeQuevedoC Thanks! Just doing more testing and I noticed that the languages are mixed up in iOS in the example. I was expecting them to be the same language from whatever the system is using. Any ideas on this?

Screen Shot 2022-02-03 at 8 04 30 AM

Screen Shot 2022-02-03 at 8 03 36 AM

**Comment by JorgeQuevedoC** [comment]
@gciluffo tbh I don't know what is causing that; but it is there since I started using the package. It just happens when simulation is on

**Review by joe307bad — APPROVED** [review]

**Comment by joe307bad** [comment]
This is epic @JorgeQuevedoC and @gciluffo. 🎉 Thanks for the work on this.

**Comment by adoniscoder** [comment]
Hey guys, thanks for all your efforts to updating to v2. I just wanted to know if there is any status on when this update will be published on npm ? because the npm's version is still 1.1.0

**Comment by gciluffo** [comment]
Hey @adoniscoder we are currently trying to get access back to the npm repository so we can publish this. The previous developers who maintained this are no longer working on this. If you need to consume the package now you can just reference like this in package.json: 

```
"@homee/react-native-mapbox-navigation": "git://github.com/homeeondemand/react-native-mapbox-navigation.git#b639f97a242e38ddf006b9c6a32b86e559049bc2",
```

**Pull request merged**